hydra/_internal/callback.py

```diff
@@ -0,0 +1,54 @@
+# Copyright (c) Facebook, Inc. and its affiliates. All Rights Reserved
+import logging
+from typing import Any
+
+from omegaconf import DictConfig
+
+from hydra.core.utils import JobReturn
+
+logger = logging.getLogger(__name__)
+
+
+class Callback:
+    def on_run_start(
+        self,
+        config: DictConfig,
+        **kwargs: Any,
+    ) -> None:
```

Comment:
1. can this be more compact if you remove the trailing comma (in all of those).
2. Add docstrings describing each event.